starknet_py/net/full_node_client.py

```diff
@@ -315,6 +319,26 @@ async def estimate_fee(
             ),
         )
 
+    async def get_block_number(self) -> int:
+        """Get the most recent accepted block number"""
+        return await self._client.call(method_name="blockNumber", params={})
+
+    async def get_block_hash_and_number(self) -> BlockHashAndNumber:
+        """Get the most recent accepted block hash and number"""
+        res = await self._client.call(method_name="blockHashAndNumber", params={})
+        return cast(BlockHashAndNumber, BlockHashAndNumberSchema().load(res))
+
+    async def get_chain_id(self) -> int:
+        """Return the currently configured Starknet chain id"""
+        return await self._client.call(method_name="chainId", params={})
+
+    async def get_syncing_status(self) -> Union[bool, SyncStatus]:
+        """Returns an object about the sync status, or false if the node is not syncing"""
+        sync_status = await self._client.call(method_name="syncing", params={})
+        if isinstance(sync_status, bool):
+            return sync_status
+        return cast(SyncStatus, SyncStatusSchema().load(sync_status))
```

Comment:
A test checking this line would be nice

Reply:
Devnet always return false, so can't do it there. I've tested it on testnet though.

Reply:
What about a mock? 👀